python/ray/data/datasource/json_datasource.py

```diff
@@ -34,6 +38,36 @@ def __init__(
 
     # TODO(ekl) The PyArrow JSON reader doesn't support streaming reads.
     def _read_stream(self, f: "pyarrow.NativeFile", path: str):
-        from pyarrow import json
+        from io import BytesIO
+
+        from pyarrow import ArrowInvalid, json
 
-        yield json.read_json(f, read_options=self.read_options, **self.arrow_json_args)
+        buffer = f.read_buffer()
+        block_size = self.read_options.block_size
+        use_threads = self.read_options.use_threads
+        max_block_size = DataContext.get_current().target_max_block_size
+        while True:
+            try:
+                yield json.read_json(
+                    BytesIO(buffer),
+                    read_options=json.ReadOptions(
+                        use_threads=use_threads, block_size=block_size
+                    ),
+                    **self.arrow_json_args,
+                )
+                break
+            except ArrowInvalid as e:
+                if (
+                    isinstance(e, ArrowInvalid)
+                    and "straddling" not in str(e)
+                    or block_size > max_block_size
+                ):
+                    raise e
```

Comment:
let's modify the error message of the exception that is raised to include the largest block size that was tried. and let's also include a link to this GH issue for more details: https://github.com/apache/arrow/issues/25674

Reply:
Updated the exception to this: `pyarrow.lib.ArrowInvalid: straddling object straddles two block boundaries (try to increase block size?) - Auto-increasing target_max_block_size to 4B failed. More information on this issue can be found here: https://github.com/apache/arrow/issues/25674`